src/OrleansAzureUtils/Storage/AzureQueueDataManager.cs

```diff
@@ -140,12 +139,12 @@ public AzureQueueDataManager(string queueName, string deploymentId, string stora
             {
                 // that way we don't have first to create the queue to be able later to delete it.
                 CloudQueue queueRef = queue ?? queueOperationsClient.GetQueueReference(QueueName);
-                var exists = Task<bool>.Factory.FromAsync(queueRef.BeginExists, queueRef.EndExists, null);
-                if (await exists)
+                if (queueRef != null)
                 {
-                    await Task.Factory.FromAsync(queueRef.BeginDelete, queueRef.EndDelete, null);
+                    await queueRef.DeleteIfExistsAsync();
                     logger.Info(ErrorCode.AzureQueue_03, "Deleted Azure Queue {0}", QueueName);
```

Comment:
only log the message if delete succeeded (the boolean value returned by DeleteIfExistsAsync)